x-pack/test/observability_ai_assistant_api_integration/tests/connectors/connectors.spec.ts

```diff
@@ -6,13 +6,18 @@
  */
 
 import expect from '@kbn/expect';
+import type { SuperTest, Test } from 'supertest';
 import { FtrProviderContext } from '../../common/ftr_provider_context';
 
 export default function ApiTest({ getService }: FtrProviderContext) {
   const observabilityAIAssistantAPIClient = getService('observabilityAIAssistantAPIClient');
   const supertest = getService('supertest');
 
   describe('List connectors', () => {
+    before(async () => {
+      await deleteAllActionConnectors(supertest);
```

Comment:
I can do that.
That being said, I've become a big fan of doing the cleanup before running the test rather than after because it makes it possible to debug broken tests by inspecting the state after test completion when things haven't been cleaned up.

Cypress docs goes into more details about this (not specific to cypress)
https://docs.cypress.io/guides/references/best-practices#Using-after-Or-afterEach-Hooks

Reply:
There are probably pros/cons with each approach. Try it out - you might like it too! :)

Reply:
but you need to do it on both presumably? I think you always want a clean slate at the end of the test

Reply:
> but you need to do it on both presumably? I think you always want a clean slate at the end of the test

If every test takes the same approach and cleans we don't need a clean after the test has run. Test contamination is only a problem if a test expects a clean slate when starting (something that's often not the case anyway). 

But I agree that in a big repo like Kibana this might not be entirely possible.

Reply:
Every test would need to the same cleanup before though for this to work (e.g., this test is clearing the connectors, but not the conversation), which I assume is a very complicated process

Reply:
A test only need to clear what is relevant for it. So if a test doesn't concern itself with connectors it doesn't need to clear connectors.